src/seedu/addressbook/AddressBook.java

```diff
@@ -431,6 +449,48 @@ private static String executeAddPerson(String commandArgs) {
     }
 
     /**
+     * Edit a person (specified by the command args) in the address book, assume a person is defind by his name
+     * The entire command arguments string is treated as a string representation of the person to add.
+     *
+     * @param commandArgs full command args string from the user
+     * @return feedback display message for the operation result
+     */
+    private static String executeEditPerson(String commandArgs) {
+        // try decoding a person from the raw args
+        final Optional<String[]> decodeResult = decodePersonFromString(commandArgs);
```

Comment:
Perhaps you can make use of SLAP here. Notice that this function is performing different levels of work.